src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolverOptionsSetup.cs

```diff
@@ -62,41 +93,192 @@ private string GetWorkersRootDirPath(IConfiguration configuration)
+        /// </summary>
         private IConfiguration GetRequiredConfiguration()
         {
-            string requiredSection = $"{RpcWorkerConstants.LanguageWorkersSectionName}:{WorkerConstants.WorkersDirectorySectionName}";
+            LogWorkersDirSectionPresence(_configuration, nameof(_configuration));
+            var configuration = _configuration;
 
+            // Use the latest configuration from the ScriptHostManager if available.
+            // After specialization, the ScriptHostManager will have the latest IConfiguration reflecting additional configuration entries added during specialization.
             if (_scriptHostManager is IServiceProvider scriptHostManagerServiceProvider)
             {
                 var latestConfiguration = scriptHostManagerServiceProvider.GetService<IConfiguration>();
-                var latestConfigValue = GetConfigurationSectionValue(latestConfiguration, nameof(latestConfiguration), requiredSection);
 
-                if (!string.IsNullOrEmpty(latestConfigValue))
+                if (latestConfiguration is not null)
                 {
-                    return latestConfiguration;
+                    LogWorkersDirSectionPresence(latestConfiguration, nameof(latestConfiguration));
+
+                    configuration = new ConfigurationBuilder()
+                        .AddConfiguration(_configuration)
+                        .AddConfiguration(latestConfiguration)
+                        .Build();
+                }
+            }
+
+            return configuration;
+        }
+
+        /// <summary>
+        /// Logs a trace message if the required configuration section is found.
+        /// </summary>
+        private void LogWorkersDirSectionPresence(IConfiguration configuration, string configurationSource)
+        {
+            string configSectionToCheck = $"{RpcWorkerConstants.LanguageWorkersSectionName}:{WorkerConstants.WorkersDirectorySectionName}";
+            var section = configuration.GetSection(configSectionToCheck);
+
+            if (!string.IsNullOrEmpty(section.Value))
+            {
+                _logger.LogTrace("Found configuration section '{requiredSection}' in '{configurationSource}'.", configSectionToCheck, configurationSource);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the list of worker probing paths from configuration or uses the default path for Windows environment.
+        /// </summary>
+        internal List<string> GetWorkerProbingPaths(IConfiguration configuration)
+        {
+            // If Configuration section is set, read probing paths from configuration.
+            var probingPathsSection = configuration.GetSection($"{RpcWorkerConstants.LanguageWorkersSectionName}:{RpcWorkerConstants.WorkerProbingPathsSectionName}");
+            var subSectionsCount = probingPathsSection.GetChildren().Count();
+            var probingPaths = new List<string>();
```

Comment:
i think we can simplify a lot of this by doing `var probingPaths = probingPathsSection.Get<List<string>>();` -- then we don't have to do any special parsing or checking or anything. And if it's null or empty, fall back to the built-in for windows.